### ISSUE TITLE
Document how to declare arrayProps of custom types

### ISSUE THREAD

**Issue by dandv** [typegoose/typegoose#228]
https://typegoose.github.io/typegoose/docs/decorators/arrayProp/#items talks about `items` being a primitive type, but what about a defined `type`? I tried using one but am getting this warning:

> `MyType' only refers to a type, but is being used as a value here

**Comment by hasezoey** [comment]
sorry, what do you mean with your question?

> `MyType' only refers to a type, but is being used as a value here

you cant use typescript types at runtime

**Comment by dandv** [comment]
Right, so the docs should say what to specify for the `items` property in the case of arrays of objects. My code incorrectly has a type there

![image](https://user-images.githubusercontent.com/33569/79635083-d5deac00-81c2-11ea-98bc-4f115b457df4.png)

Should I define and pass a `Link` class instead?

**Comment by hasezoey** [comment]
you cant use typescript types at runtime, only classes - so yes, define them as classes

**Comment by dandv** [comment]
Defining `items: Link` did not work - mongoose returns null for both fields, `url` and `text`.

If I declare `@arrayProp({ items: Object }) links: Link[];` instead, I see the `Implicitly setting "Mixed" is not allowed!` warning, but I get the correct result (`url` and `text` containing the correct strings from the document in the collection).

My code is as in the screenshot:

```ts
class Link {
  url: string;
  text?: string;
}

export class Discussion extends defaultClasses.TimeStamps {
  @arrayProp({ items: Link }) links: Link[];
}
// WRONG: link.url and link.text are null
const link = await DiscussionModel.findOne({ _id: id });
```

**Comment by hasezoey** [comment]
@dandv take a look at what i modified on your (now merged) pr

TL;DR: add `@prop` to your subdocuments properties
